Tests/MongoSwiftSyncTests/SyncChangeStreamTests.swift

```diff
@@ -747,14 +801,22 @@ final class SyncChangeStreamTests: MongoSwiftTestCase {
     }
 
     func testChangeStreamOnADatabase() throws {
-        guard MongoSwiftTestCase.topologyType != .single else {
-            print(unsupportedTopologyMessage(testName: self.name))
-            return
-        }
-
         let client = try MongoClient.makeTestClient()
-        guard try client.serverVersion() >= ServerVersion(major: 4, minor: 0) else {
-            print("Skipping test case for server version \(try client.serverVersion())")
+        let testRequirements = TestRequirement(
+            minServerVersion: ServerVersion(major: 4, minor: 0),
+            acceptableTopologies: ["replicaSet", "sharded"].map { TestTopologyConfiguration(from: $0) }
+        )
+
+        let unmetRequirements = try client.meetsRequirements(testRequirements)
+        guard unmetRequirements == nil else {
+            switch unmetRequirements {
+            case .minServerVersion:
+                print("Skipping test case for server version \(try client.serverVersion())")
```

Comment:
I think the message from `printRequirementNotMetMessage` can be used instead of this

Reply:
done